src/metadata/option-set-query-key-factory.js

```diff
@@ -0,0 +1,44 @@
+const optionSetFields = 'id,version,options[id,displayFormName]'
+
+const queryKeyFactory = {
```

Comment:
The way I understand the current architecture for query key factories is: There should be a `query-key-factory.js` in the root of the module (this file is already at the root of metadata, so the location is fine, but the name is not). And that file can export multiple things, like the [`data-workspace/query-key-factory.js`](https://github.com/dhis2/data-entry-app/blob/development/src/data-workspace/query-key-factory.js).

Not sure whether that's the best approach, but it's the current "standard" which @ismay landed at after reading through the `react-query`'s author's blog posts.

Reply:
Are you referring to the prefix `option-set`? If so I can rename it, It was just more specific... 
Not sure what else you are referring to? This does not need to return "multiple" things. It's all just the same API-endpoint, and thus the only object needed. I feel like the structure of the actual factory complies to the structure of the other factories? Just that this only operates on the "optionSet"-endpoint, and thus all methods are related?

I guess you could argue that the `allVersion`-one is a "separate"-entity, and thus could be it's own object. But I decided to keep it simple, and "group" them by endpoint. If that is what you are referring to, we can change it. But if the structure is somehow otherwise different than the other factories or the link provided by @ismay please explain.

Reply:
btw. I think I followed the style in `query-key-factory` in custom-forms. Which exports the `queryKeyFactory` as a default export.